examples/aa-simple-dapp/src/hooks/usePlugin.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  UninstallPluginParams,
+  accountLoupeDecorators,
+  pluginManagerDecorator,
+  type IMSCA,
+  type Plugin
+} from "@alchemy/aa-accounts";
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { Address, UserOperationOverrides } from "@alchemy/aa-core";
+import { useCallback, useMemo } from "react";
+
+export const usePlugin = <AD, PD>(
+  provider: AlchemyProvider,
+  plugin: Plugin<AD, PD>,
+  pluginAddress_?: Address,
+) => {
+  if (!provider.isConnected<IMSCA>()) {
+    throw new Error(
+      "Provider must be connected to an MSCA before using this hook!"
+    );
+  }
+
+  const {
+    accountMethods,
+    providerMethods,
+    meta: { addresses },
+  } = plugin;
+
+  const pluginAddress = useMemo(() => pluginAddress_ ?? addresses[provider.rpcClient.chain.id], [pluginAddress_, addresses, provider]);
```

Comment:
```suggestion
  const _pluginAddress = useMemo(() => pluginAddress ?? addresses[provider.rpcClient.chain.id], [pluginAddress, addresses, provider]);
```

Reply:
I think we should keep this as is. the `pluginAddress_` is the potential override. The `pluginAddress` that's meant to be used throughout is actually either this override or the one from the `meta` field. We follow this same pattern when parsing parsing configs with zod in other places